cxx/hirm.hh

```diff
@@ -537,11 +542,16 @@ class Relation {
   Relation(const Relation&) = delete;
 };
 
+using RelationVariant =
```

Comment:
We should add a note in distributions/, maybe in distributions/base.hh, telling people to add their distribution to this list if they want it to be used.

We should also add a note here telling people to add their distribution's SampleType to the ObservationVariant list in this file.  It might be worth moving this definition up to be next to ObservationVariant's definition for that reason.